### PR TITLE
migration fixes

### PR DESCRIPTION
**Pull request by debris**
- osx: limit numer of database open files during backup
- linux: do not use temp directory

### CODE REVIEW

File: parity/migration.rs [hunk 1]
```diff
@@ -144,26 +153,40 @@ fn migrate_database(version: u32, path: PathBuf, migrations: MigrationManager) -
 
 	println!("Migrating database {} from version {} to {}", path.to_string_lossy(), version, CURRENT_VERSION);
 
-	// get temp path
-	let temp_path = temp_database_path();
+	let temp_path = temp_database_path(&path);
+	let backup_path = backup_database_path(&path);
 	// remote the dir if it exists
 	let _ = fs::remove_dir_all(&temp_path);
+	let _ = fs::remove_dir_all(&backup_path);
```
Comment (tomusdrw) [inline parity/migration.rs]: Shouldn't you use a lock file of some kind? If someone accidentaly runs parity the second time during migration?

File: util/src/journaldb/archivedb.rs [hunk 2]
```diff
@@ -47,7 +47,9 @@ impl ArchiveDB {
 	/// Create a new instance from file
 	pub fn new(path: &str) -> ArchiveDB {
 		let opts = DatabaseConfig {
-			prefix_size: Some(12) //use 12 bytes as prefix, this must match account_db prefix
+			//use 12 bytes as prefix, this must match account_db prefix
+			prefix_size: Some(12),
```
Comment (tomusdrw) [inline util/src/journaldb/archivedb.rs]: Why not a constant?

File: parity/migration.rs [hunk 3]
```diff
@@ -144,26 +153,40 @@ fn migrate_database(version: u32, path: PathBuf, migrations: MigrationManager) -
...
 
 	{
+		let db_config = DatabaseConfig {
+			prefix_size: None,
+			max_open_files: 64,
+		};
+
 		// open old database
-		let old = try!(Database::open_default(path.to_str().unwrap()).map_err(|_| Error::MigrationFailed));
+		let old = try!(Database::open(&db_config, path.to_str().unwrap()).map_err(|_| Error::MigrationFailed));
```
Comment (rphmeier) [inline parity/migration.rs]: maybe have MigrationFailed store an error so we don't discard error information?

File: util/src/journaldb/archivedb.rs (hunk 2 above)
Comment (debris) [inline util/src/journaldb/archivedb.rs]: agree, but it's not a part of this pr

Comment (tomusdrw) [inline util/src/journaldb/archivedb.rs]: Sure :)

**Comment by tomusdrw** [comment]
All the comments are not concerning the fix that this PR is about. I think we should merge that and address the concerns in separate PR.

File: parity/migration.rs (hunk 1 above)
Comment (debris) [inline parity/migration.rs]: Agree, but it's unlikely to happen. Let's keep it as it is for now. Afaik, there is no cross platform (including windows) library for locking file. I've opened an issue not to forget about this: https://github.com/ethcore/parity/issues/1156

**Pull request merged**